src/time-picker/TimePicker.js

```diff
@@ -0,0 +1,211 @@
+import React from 'react';
+import { props, t, skinnable } from '../utils';
+import cx from 'classnames';
+import Dropdown from '../dropdown/Dropdown';
+import range from 'lodash/utility/range';
+import flatten from 'lodash/array/flatten';
+
+/*
+  Options is defined here cause we need to not change the ref.
+  This happend because of a very wired behaviour of react-select v0.6.12
+*/
+const options = [];
+
+const H24 = '24h';
+const H12 = '12h';
+const symbolRegex = /[\s,\.\|\/\\]/;
+const numberRegex = /^\d+$/;
+const time12Regex = /^([1-9]|1[0-2])[:,\.\|\/\\]?([0-5]\d)?\s*?(AM|PM)$/;
+const time24Regex = /^([01]?\d|2[0-3])[:,\.\|\/\\]?([0-5]?\d)?$/;
+const separator = ':';
+
+const pad = (num) => num <= 9 ? `0${num}` : num;
+const timeFormatter24 = (hour, minute) => `${pad(hour)}:${pad(minute)}`;
+const timeFormatter12 = (hour, minute) => {
+  if (hour === 0 || (hour > 12 && hour % 12 === 0)) {
+    return `12${separator}${pad(minute)} am`;
+  } else if (hour === 12) {
+    return `12${separator}${pad(minute)} pm`;
+  } else if (hour >= 13) {
+    return `${pad(hour % 12)}${separator}${pad(minute)} pm`;
+  } else {
+    return `${pad(hour)}${separator}${pad(minute)} am`;
+  }
+};
+
+const isValidTime = input => time12Regex.test(input) || time24Regex.test(input);
+
+const formatValue = (timeFormatter, hour, minute) => ({
+  value: `${hour}${separator}${minute}`,
+  label: timeFormatter(hour, minute)
+});
+
+const lteTime = (minTime, maxTime) => (
+  maxTime.hours > minTime.hours || (maxTime.hours === minTime.hours && maxTime.minutes >= minTime.minutes)
+);
+
+const addZero = str => {
```

Comment:
This function add a `0` at the end of every string that is already been cleaned and it is in the format `HH:m` in any other cases it will do nothing at all. It will be called just at the very end, and is useful to manage the decision we make to add a `0` when we are into an undefined situation like that. We can say that this is a "By Design behaviour"